.github/workflows/tests.yml

```diff
@@ -132,11 +131,80 @@ jobs:
           path: ~/.m2/repository
           key: ${{ runner.os }}-${{ steps.get-date.outputs.date }}-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-maven-
-
       - name: Use silent log config
         run: mv chore/logback.xml src/test/resources/
-      - name: Run extra checks
-        run: ./chore/ci-extra.sh
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          fetch-depth: 0
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y python3-pip
+          pip3 install --user CommonMark==0.9.1 requests pygithub
+
+      - name: Verify and Site Maven goals
+        run: |
+          mvn verify license:check site install -DskipTests -DadditionalJOption=-Xdoclint:syntax,-missing -Dscan
+          cd spoon-pom && mvn install -DskipTests && cd ..
+
+      - name: Checkstyle in src/tests
+        run: mvn -q checkstyle:checkstyle -Pcheckstyle-test
+
+      - name: Check documentation links
+        run: python3 ./chore/check-links-in-doc.py
+
+      - name: Analyze dependencies through DepClean in spoon-core
+        run: mvn -q depclean:depclean
+
+      - name: Spoon-decompiler
+        run: |
+          cd spoon-decompiler
```

Comment:
How about using the `working-directory` step option for such things: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun?

Reply:
I convert this. Also I remove the bash script